hack/cmd_util.sh

```diff
@@ -322,6 +323,76 @@ function os::cmd::internal::get_results() {
 	cat "${os_cmd_internal_tmpout}" "${os_cmd_internal_tmperr}"
 }
 
+# os::cmd::internal::get_try_until_results returns a concise view of the stdout and stderr output files
+# using a timeline format, where consecutive output lines that are the same are condensed into one line
+# with a counter
+function os::cmd::internal::print_try_until_results() {
+	if grep -vq "^====BETWEEN-ATTEMPTS====$" "${os_cmd_internal_tmpout}"; then 
+		echo "Standard output from the command:"
+		os::cmd::internal::compress_output "${os_cmd_internal_tmpout}"
+	else 
+		echo "There was no output from the command."                                      																																																																																																																
+	fi	
+
+	if grep -vq "^====BETWEEN-ATTEMPTS====$" "${os_cmd_internal_tmperr}"; then 
+		echo "Standard error from the command:"
+		os::cmd::internal::compress_output "${os_cmd_internal_tmperr}"
+	else 
+		echo "There was no error output from the command."                                      																																																																																																																
+	fi
+}
+
+# os::cmd::internal::mark_attempt marks the end of an attempt in the stdout and stderr log files
+# this is used to make the try_until_* output more concise
+function os::cmd::internal::mark_attempt() {
+	echo "====BETWEEN-ATTEMPTS====" >> "${os_cmd_internal_tmpout}" | tee "${os_cmd_internal_tmperr}"
+}
+
+# os::cmd::internal::compress_output compresses an output file into timeline representation
+function os::cmd::internal::compress_output() {
+	local logfile=$1
+
```

Comment:
> Understood. If I do any further work on that it will be strictly on my own time.

It's not just about your time.  This isn't likely to grow or extend like the work we had in the commit scrubber that Dan worked on.  This simply isn't a good candidate for a helper utility.  We've got tests to make sure these functions keep working for minor maintenance, but this is code that's essentially "done".  If someone maintaining has to drop into a helper program, they're just going to hate you.